File: inspect_db_duplicates.mjs
import Database from 'better-sqlite3';
import path from 'path';
import os from 'os';

const dbPath = path.join(process.env.APPDATA, 'AgriLedger', 'data', 'agridb.db');
const db = new Database(dbPath, { readonly: true });

console.log('DB Path:', dbPath);
const total = db.prepare('SELECT COUNT(*) AS cnt FROM purchases').get().cnt;
console.log('Total purchases:', total);

const dupRows = db.prepare(`
  SELECT date, supplier_name, receipt_number, expense_category, gross_amount, COUNT(*) AS cnt
  FROM purchases
  GROUP BY date, supplier_name, receipt_number, expense_category, gross_amount
  HAVING cnt > 1
  ORDER BY cnt DESC, expense_category
  LIMIT 50
`).all();
console.log('Duplicate groups (same date/supplier/receipt/category/gross):', dupRows.length);
dupRows.forEach((row) => {
  console.log(row);
});

const sumByCategory = db.prepare(`
  SELECT expense_category AS category, COUNT(*) AS cnt, SUM(gross_amount) AS total
  FROM purchases
  GROUP BY expense_category
  ORDER BY total DESC
`).all();
console.log('\nCategory summary:');
sumByCategory.forEach(r => console.log(r));

File: inspect_db_duplicates.py
import os
import sqlite3
from pathlib import Path

appdata = os.getenv('APPDATA')
if not appdata:
    raise SystemExit('APPDATA env var missing')

path = Path(appdata) / 'AgriLedger' / 'data' / 'agridb.db'
print('DB PATH:', path)
conn = sqlite3.connect(path)
cur = conn.cursor()
cur.execute('SELECT COUNT(*) FROM purchases')
print('Total purchases:', cur.fetchone()[0])
cur.execute('''
SELECT date, supplier_name, receipt_number, expense_category, gross_amount, COUNT(*) as cnt
FROM purchases
GROUP BY date, supplier_name, receipt_number, expense_category, gross_amount
HAVING cnt > 1
ORDER BY cnt DESC, expense_category
LIMIT 50
''')
dups = cur.fetchall()
print('Duplicate groups:', len(dups))
for row in dups:
    print(row)
cur.execute('''
SELECT expense_category, COUNT(*) as cnt, SUM(gross_amount) as total
FROM purchases
GROUP BY expense_category
ORDER BY total DESC
''')
print('--- category totals ---')
for row in cur.fetchall():
    print(row)
conn.close()
